### ISSUE TITLE
Setting up a "store"

### ISSUE THREAD

**Issue by rinigus** [sailfishos-chum/main#29]
As integration with OpenRepos does not seem to be happening at any significant speed, we should consider backing up Chum repos with some kind of experience similar to app stores. This issue is made to discuss it and agree on steps forward.

Ideally, we need an app to interact with Chum on device and some kind of web presence. I'll try to write up what would such experience could be and later some implementation aspects. In principle, we should aim for something similar to Storeman and OpenRepos.

**App:**

- [x] list available software in categories/subcategories
- [x] search available software
- [x] install software
- [x] remove software
- [x] see recent updates
- [x] each software package could have icon, screenshots, issues and discussion forums, some stats.

**Web:**

Ideally same as for app, in part that makes sense:
- [ ] list available software in categories/subcategories
- [x] search available software
- [ ] see recent updates
- [ ] each software package could have icon, screenshots, issues and discussion forums, some stats.

@mentaljam can surely add to that list or comment on it.

If we start from the app and can provide all data for that, maybe it will be easier to roll out web interface later. In the current state, sailfishos-chum-gui already shows that it is possible to get list, some info on app, install and remove it. 

Apps are missing icons, issues, and so on. In principle, we can add such data and provide to ChumGui via some files online. For example, each app can add at OBS some special file `meta.yaml` that would list app's repo, link to issues, links to discussion forums (multiple if needed), link to releases, link to icon. Unless we can specify in SPEC and make it available to ChumGui from PackageKit. Those package-specific meta.yaml datasets can be collected by a crawler and published at some github repo or in some S3 type bucket. This would allow ChumGui to refresh it as soon as needed from a single source. The crawler can just update it (if needed) once an hour, for example.

For GitHub and GitLab repos, we can also show number of stars (and forks), if some simple API is available. That would indicate the status of the app similar to currently used stars in OpenRepos. Advantage of guiding users to online issue trackers and discussion forums is a great experience for developers. In addition, it allows to merge users from different platforms for the apps that are multi-platform.

Thoughts, ideas? 

cc: @mentaljam @piggz

**Comment by rinigus** [comment]
I would like resurrect this topic and proceed with it. As we have no feedback on what's possible from OBS side, let's implement metadata separately and rely on PackageKit or libzypp on processing RPM data.

At Chum, we have a mix of libraries and applications. For the purpose of the store, I propose that each package that would like to be exposed in Chum GUI, would have to define metadata file with the relevant information. The ones missing this info, will be not shown in GUI. For example, there is no point in exposing support libs to GUI as users don't really need to install it separately. Unless they develop something and then can do it via command line.

As this metadata file has to be written by a developer, I propose to use YAML for it. For Pure Maps, it would be
```YAML
name: Pure Maps
package: harbour-pure-maps
category: Applications/Maps
repotype: github
repo: https://github.com/rinigus/pure-maps
icon: https://raw.githubusercontent.com/rinigus/pure-maps/master/data/pure-maps.svg
screenshots:
  - https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/main.png
  - https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/menu.png
  - https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/traffic.png
forums:
  - https://github.com/rinigus/pure-maps/discussions
issues: https://github.com/rinigus/pure-maps/issues
```

This file defines icon, screenshots and relevant links (forum together with issues). Forum is kept in plural to allow support for FSO, TMO and other locations. Issues are usually in central location, hence kept as a single item. It is expected, that metadata file would define `package` key for which it is applicable. Category has to be defined from the given list and separate from SPEC, as in SPEC, at least according to Fedora, it is deprecated. I have also added `repo` together with `repotype` which would allow us to load releases, issues, stars by using GitHub or GitLab API.

In principle, one package (SPEC) can define multiple RPMs and we may wish to have support for it. In YAML, we can separate and have multiple documents using `---` as a separating line.

After processing all packages metadata, we would end up with JSON which is easy to load on Chum GUI side. For example, as in 
```JSON
[
    {
        "name": "Pure Maps",
        "package": "harbour-pure-maps",
        "category": "Applications/Maps",
        "repotype": "github",
        "repo": "https://github.com/rinigus/pure-maps",
        "icon": "https://raw.githubusercontent.com/rinigus/pure-maps/master/data/pure-maps.svg",
        "screenshots": [
            "https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/main.png",
            "https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/menu.png",
            "https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/traffic.png"
        ],
        "forums": [
            "https://github.com/rinigus/pure-maps/discussions"
        ],
        "issues": "https://github.com/rinigus/pure-maps/issues"
    }
]
```

There are several ways we can now generate and distribute these files. One option is to have them at OBS as one of the package sources (named `chum.yaml` or similar). Alternative, would be to make dedicated GitHub project and add those via PRs. Latter would allow to categorize and drop `package` keyword as well through use of subdirectories and corresponding metadata filenames. For example, GitHub project would have a file

`packages/applications/maps/harbour-pure-maps.yaml`

All packages would be combined into a single JSON and put into that repository allowing Chum GUI to pull the file from GitHub.

I would prefer to have metadata for each package and overall generated JSON in the same GitHub repo and avoid OBS+Github combination. It would allow us to just regenerate such store definition file on each import and not to hunt down changes in OBS. Changes in OBS can be tracked automatically, if needed, but it is not very simple and may require something checking it periodically.

Thoughts / comments?

**Comment by poetaster** [comment]
I like this approach. The only thing I'm not too keen about is aggregating in a central repository.  I suspect keeping things in sync might get hairy.  I was thinking a lazy man's solution would simply use the .spec's Description field. So, if that field contains, for instance subset:

```
Description    : 
    View maps, find places and routes, navigate with turn-by-turn instructions,
    search for nearby places by type and share your location.

    repo: https://github.com/rinigus/pure-maps
    icon: https://raw.githubusercontent.com/rinigus/pure-maps/master/data/pure-maps.svg
    screenshots: 
      - https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/main.png
      - https://raw.githubusercontent.com/rinigus/pure-maps/master/screenshots/menu.png
    forums:
      - https://github.com/rinigus/pure-maps/discussions
    issues: https://github.com/rinigus/pure-maps/issues

```
So, stick to the yaml idea but first just in the spec? I'm not sure this is even feasible.

**Comment by rinigus** [comment]
If we add YAML into description field, there are few disadvantages:

- all will have to make new releases to trigger the builds
- this Chum specific info will get distributed all over the place and will be a part of package description in all stores (Jolla and OpenRepos) and, if you happen to work on other Linux distros with SPEC (Fedora and SuSe), them as well. Although, that part of description can be probably protected by some macros.

Notice that this YAML doesn't have a release info. Which, in practice, means that we have to make aggregation once in a while. 

Advantage of having it in a SPEC would be easier control by developers in a longer run. For example, change links, add screenshots and such. It would also not need any central aggregation as it seems to be easy to generate package model from just parsing description for additional info. 

So, to summarize, maybe adding it to SPEC would be better in longer term.

**Comment by poetaster** [comment]
Ah, I hadn't thought of the first point. But I have a bunch of bug releases ;) Fair point.

Yeah, it did think it's a bit much if you also package for other distros. On the other hand, spec files permit conditionals. It does feel a bit spammy. On the other hand, it's quite descriptive :)

My thinking was, make a yaml stanza under description to start experimenting with what it might look like. Logic like sorting on Category and QML for displays can use the same field mappings even if later we see a need to move the data.

It does feel like a hack. I have to admit to liking hacks more than is healthy.

**Comment by rinigus** [comment]
From discussion at IRC:

- we would like to add that metadata into description of SPEC. 
- it is possible to use `%if` in description to drop out that metadata in Linux distributions or SFOS distribution channels that don't use it. it is up to developers to decide
- for apps that don't supply this info, minimal info is displayed as it is now. I guess under "All packages" category

So, I suggest to have then %description in the format

```SPEC

%description
This is my awesome app

This is continuation of regular app description

name: Awesome App
category: Applications/Misc
...
```

So, the last paragraph of description would be parsed for YAML. Probably the other paragraphs should be processed for word alignment as well, using regular approach.

I will look into writing %description parser, hopefully tonight

**Comment by piggz** [comment]
Here is some progress on this.
![JwSNShUbIfIiHcNDIQqpXAXY](https://user-images.githubusercontent.com/4650063/144932569-33a743fb-34a0-4788-bdb3-10df2cf679a7.png)
Current code is https://github.com/piggz/sailfishos-chum-gui/tree/metadata
And an example .spec is https://github.com/piggz/harbour-advanced-camera/blob/metadata/rpm/harbour-advanced-camera.spec

**Comment by piggz** [comment]
I borrowed a few components from storeman, hope thats ok @mentaljam

**Comment by mentaljam** [comment]
Hi!

Sorry if missed, what's the benefit of using the description field of .spec? That a developer doesn't need to push some additional files to the Chum GitHub repo?

**Comment by mentaljam** [comment]
> I borrowed a few components from storeman, hope thats ok @mentaljam

Which I've borrowed from the official store app😉

**Comment by rinigus** [comment]
Looks nice! 

Re keeping it in SPEC: it will simplify updates of the fields for developers. Other benefits are:
- it avoids a nasty problem where you have to start matching RPMs to app descriptions. Same SPEC can define multiple packages, so it may become non-trivial
- as RPM carries its own description, user will get only apps that (s)he has in the defined repo. As Chum apps don't always support all arch/SFOS release combinations, we get only relevant content automatically.

@piggz: I really love the fact that we have pulley menu working again :) . 

Note that for Github and Gitlab we can detect many fields automatically as they have corresponding API. See my comments from earlier discussion over here regarding it. So, for those, we can even skip adding issues link in SPEC. But maybe we should start filing separate issues under chum-gui regarding design specifics.

As for `BEGINCHUMMETADATA`, let's avoid it. Instead, let's parse the last paragraph for Chum data in description. Note that you can avoid `package` field - we don't need to match package to metadata anymore. For GitHub, `repotype` can be dropped as that we can deduce automatically. 

What about the list of packages? Have you had a chance to work on those?

Maybe you could push the changes into a new branch of chum-gui?

I'll be able to join you tonight, I hope. So, let me know where I should start. May I take Github API parsing, for example?

**Comment by rinigus** [comment]
@piggz: Just saw that you convert YAML->JSON string and parse it in QML. As I'd like to add more integration with Github/Gitlab (such as showing current issues from there, release info for changelogs), it would make sense to parse it already in C++ and export data as properties. For example, "name" would be used to show package in the list and not the name of RPM. Similar exposure is expected for icon, number of stars in Github/Gitlab.

**Comment by Thaodan** [comment]
Would it make sense to use appstream as backend for metadata? Its already in a similar manager in gnome and kde.
https://www.freedesktop.org/wiki/Distributions/AppStream/

**Comment by rinigus** [comment]
After some discussion, we ended by preferring adding metadata to SPEC description. See above for summary and pros/cons of that approach. In SPEC, we can, in principle have a link to appstream XML and load from there. However, I just remember how painful it was to fill it. See example of Pure Maps: https://github.com/rinigus/pure-maps/blob/master/packaging/pure-maps.appdata.xml . When you compare it with the examples above and the one used by @piggz, you could clearly see the difference. As we ask developers to fill it in, I would expect many would be happier to do it with SPEC extra than with appstream. Which would hopefully increase % of packages having such metadata.

**Comment by Thaodan** [comment]
I understand. Just asking did you try to generate the appstream metadata from rpm?
See here:
https://github.com/hughsie/appstream-glib#appstream-builder

**Comment by Thaodan** [comment]
Also appstream can use a yaml backend instead of xml.

**Comment by Thaodan** [comment]
https://docs.fedoraproject.org/en-US/packaging-guidelines/AppData/#_metainfo_xml_file_creation

> When you compare it with the examples above and the one used by @piggz, you could clearly see the difference. As we ask developers to fill it in, I would expect many would be happier to do it with SPEC extra than with appstream.

If I compare the example here to one  the Fedora the difference is mainly XML vs YAML format.

**Comment by mentaljam** [comment]
@Thaodan, does appstream-builder support additional tags in RPM .spec? I don't know much about AppStream but fast googling makes me think that developers have to write additional metadata files in any case. For example, I didn't find a way how appstream-builder can get screenshots from a .spec file (searched [here](https://github.com/hughsie/appstream-glib/blob/master/libappstream-builder/asb-package-rpm.c)). On the other hand, if YAML is supported, it seems to me that it doesn't matter where you put the additional metadata: in the .spec or in the appstream file.

But the main disadvantage is that we return to the model with some kind of crawler for building the appstream metadata for OBS repositories.

**Comment by rinigus** [comment]
> I understand. Just asking did you try to generate the appstream metadata from rpm?

I don't remember doing it. Just note that we need data which is in addition to the one provided by RPM SPEC. So, while we can generate base XML (assuming that it works), it will have to be edited by developers further.

> Also appstream can use a yaml backend instead of xml

I wasn't aware of it (https://www.freedesktop.org/software/appstream/docs/sect-AppStream-YAML.html#spec-dep11-example). In that example we have multiple apps, so only one section would have to be considered. I suspect that AppStream libs will fail to load if you don't provide some minimal info which, in part, maybe duplicating what is in SPEC already.

Now, we can support the both ways as well - instead of adding metadata in specified YAML spec, provide AppStream link. I just suspect, based on my experience of filling it, not many will be thrilled doing so. So, I'd prefer to have that little metadata section in SPEC and accept any PR coming our way that adds AppStream link support. :)

**Comment by poetaster** [comment]
I'd like to suggest that we simply consider using the YAML field naming conventions from appstream but stick to first using the spec files. I've been doing a bunch of communicating with other developers and think it would be great, for the start  to keep the barriers low.

**Comment by piggz** [comment]
So, that would be some subset of https://www.freedesktop.org/software/appstream/docs/sect-AppStream-YAML.html#spec-dep11-fields ?

**Comment by Thaodan** [comment]
OBS also seem to support Appstream:
https://en.opensuse.org/openSUSE:AppStore

**Comment by rinigus** [comment]
The devil is in the details. as @poetaster, I would keep the required info to minimal and as simple to enter as possible. Again, if OBS supports AppStream and there are developers wishing to use - sure, why not. Just someone would have to research the area and write the code. Use of AppStream parsers on partially filled data will most probably fail. 

Note that when we have all data loaded from SPEC description field and all code working, it is possible to extend or convert to AppStream iff it will be found advantageous. At this stage, we have a bare-bones GUI that could really use some development. Taking into account the constraints, as discussed above, adding fields to SPEC description field looks to be the simplest way forward.

**Comment by rinigus** [comment]
@piggz: I guess we could use some subset. They probably made some research before formalizing it and we can rely on it.

**Comment by piggz** [comment]
I pushed an update which uses the last paragraph method.

I also went through the appstream yaml, and tweaked the format @rinigus proposed into this:
    name: Advanced Camera
    package: harbour-advanced-camera
    type: desktop-application
    developername: Adam Pigg
    categories: 
     - Media
     - Video
    custom:
     - repotype: github
     - repo: https://github.com/piggz/harbour-advanced-camera
    icon: https://raw.githubusercontent.com/piggz/harbour-advanced-camera/master/harbour-advanced-camera.svg
    screenshots: 
     - https://github.com/piggz/harbour-advanced-camera/raw/metadata/screenshots/screenshot1.png
     - https://github.com/piggz/harbour-advanced-camera/raw/metadata/screenshots/screenshot2.png
     - https://github.com/piggz/harbour-advanced-camera/raw/metadata/screenshots/screenshot3.png
    url:
     - homepage: https://github.com/piggz/harbour-advanced-camera
     - help: https://github.com/piggz/harbour-advanced-camera/discussions
     - bugtracker: https://github.com/piggz/harbour-advanced-camera/issues
     - donation: https://www.paypal.me/piggz

Im still open to doing it either in the description, or, if OBS can generate the data easily, using that, but id like to see a demo of that.  Perhaps phase 2?

**Comment by poetaster** [comment]
Ok. I was about to start implementing so it's good to know, uh, where. So, last stanza of Spec? 

And a small nitpick: the field names in the appstream spec. are Name case. Should I use Name case? 

And one last question, repotype is not really about type, that would be darcs, svn, git. So, we require git. Presumably we'll support gitea/gogs ?

**Comment by rinigus** [comment]
@piggz, looks good! Thanks for pushing it into local branch. 

I was looking into GitHub GraphQL API and will base Github support on that. It is rather handy and allows us to fetch exactly the data that we need and probably via single call per package page, without rather verbose extras as we get in REST API.  The sticky point is that we need to provide a token for accessing that API, even in read-only. However, I was thinking of creating a dedicated GitHub user and create a token for that. Similar API is available for GitLab. When I get to GitLab, we will see if we can support self-hosted instances - would depend on token requirements.

Looking at changed format, it seems to be fine. We don't need `package` field at all, as it is associated via SPEC already. We don't need `repotype: github` as that can be deduced from URL of `repo`. In case of GitHub (and probably GitLab), your example could be reduced to 

```YAML
name: Advanced Camera
type: desktop-application # not sure we need that
categories:
- Media
- Video
custom:
- repo: https://github.com/piggz/harbour-advanced-camera
icon: https://raw.githubusercontent.com/piggz/harbour-advanced-camera/master/harbour-advanced-camera.svg
screenshots:
- https://github.com/piggz/harbour-advanced-camera/raw/metadata/screenshots/screenshot1.png
- https://github.com/piggz/harbour-advanced-camera/raw/metadata/screenshots/screenshot2.png
- https://github.com/piggz/harbour-advanced-camera/raw/metadata/screenshots/screenshot3.png
url:
- donation: https://www.paypal.me/piggz
```

Developer Name (GitHub handle and probably full name) can be obtained from GitHub, homepage can be set in GitHub and could be either GitHub repo or any arbitrary page (see Pure Maps); issues have standard location; help can be set to discussions automatically, donation maybe as well, but I am not sure.

I would propose to load all data automatically from GitHub/GitLab and then merge it with provided YAML description taking into account that YAML-provided data overwrites automatic one. I am currently considering having an abstract package class with sub-classes for GitHub, GitLab, plain YAML implementations. That will allow to expand it later to AppStream as well. Will hopefully start on coding tonight...

**Comment by poetaster** [comment]
Which brings me back to my last question which I would now switch to a recommendation. Let's not use apis we don't NEED to use. I've probably mised something, but what is it we need to fetch from the repos if we already have all the meta data in the spec?

**Comment by rinigus** [comment]
> And one last question, repotype is not really about type, that would be darcs, svn, git. So, we require git. Presumably we'll support gitea/gogs ?

it is more about API that allows to fetch info regarding a project. 

As for need: it would allow to show Releases info and current issues. We can also show GitHub/GitLab forks, stars, and such. Whether it is an overkill, I don't know.

As for self-hosted: do you plan GitLab like? if so, this should work as long as you give us a token (not sure if it needed, will have to experiment first).

**Comment by mentaljam** [comment]
> We don't need `repotype: github` as that can be deduced from URL of `repo`.

What about self-hosted GitLab instances?

**Comment by rinigus** [comment]
@mentaljam : self-hosted would need it, indeed. Assuming we can get queries working on those - may depend on token availability. If token is required, we would have to add that together with URL base to support each self-hosted GitLab separately.

Right now I don't know how many apps we have in self-hosted GitLab instances. For Chum, I suspect it is zero or small number - at least nothing jumps into my mind.

**Comment by piggz** [comment]
@rinigus fine to have a minimal yaml, but id maybe keep some optional fields (was going to add that later) so that eg, if help is provided on a sfos forum post, that can be linked instead of detecting eg github discussions url.

also agree with @poetaster to use CamelCase

**Comment by rinigus** [comment]
> Ok. I was about to start implementing so it's good to know, uh, where. So, last stanza of Spec?

@poetaster, see https://github.com/piggz/harbour-advanced-camera/blob/metadata/rpm/harbour-advanced-camera.spec#L37

**Comment by mentaljam** [comment]
Then we may think about rather strict requirements for repositories: only git at GitHub or central GitLab. If someone prefer to use another VCS or another web-service for it they could use a separate GitHub/GitLab repo for storing packaging rules and metadata for Chum. In any case I don't think there would be much users who don't use GitHub/GitLab.

**Comment by poetaster** [comment]
> > And one last question, repotype is not really about type, that would be darcs, svn, git. So, we require git. Presumably we'll support gitea/gogs ?
> 
> it is more about API that allows to fetch info regarding a project.
> 
> As for need: it would allow to show Releases info and current issues. We can also show GitHub/GitLab forks, stars, and such. Whether it is an overkill, I don't know.
> 
> As for self-hosted: do you plan GitLab like? if so, this should work as long as you give us a token (not sure if it needed, will have to experiment first).

I run gogs and gitea instances.  I still don't understand what data we need. Nothing against GraphQL ;)

**Comment by poetaster** [comment]
> > Ok. I was about to start implementing so it's good to know, uh, where. So, last stanza of Spec?
> 
> @poetaster, see https://github.com/piggz/harbour-advanced-camera/blob/metadata/rpm/harbour-advanced-camera.spec#L37

Ok, so still under Description with conditional with newline between. Thanks!

**Comment by rinigus** [comment]
@mentaljam : git / svn  and so on don't really matter. Plan is to use API that GitHub and GitLab provide to enrich user experience. For packages/apps that cannot do that, we can always use metadata provided via SPEC %description (and/or AppStream). As such, not strict requirement - we can always fallback to data we have.

**Comment by poetaster** [comment]
@rinigus Ok. The need is not a need. But then, alternate app stores is kind of one of humanities big problems :).

**Comment by poetaster** [comment]
Ok. I've updated 2 thus far. Im going to push the releases on chum:testing . These:
https://github.com/poetaster/harbour-beamdimension/blob/main/rpm/harbour-beamdimension.spec
https://github.com/poetaster/harbour-concretemixer/blob/main/rpm/harbour-concretemixer.spec
work?

**Comment by rinigus** [comment]
Please drop "Package" field, it is not needed at all. If we keep it in examples, it will later will be adopted by all... :(

**Comment by poetaster** [comment]
Yo. Sadly, these don't build.

**Comment by rinigus** [comment]
As discussed at IRC. `tar_git` misreads SPEC and picks up 'Name' in Chum description section. Something a proper RPM SPEC parsing does not (as rpmspec, for example). To avoid clashing with tar_git, we should use some other keyword instead of name. mal suggested PackageName

**Comment by black-sheep-dev** [comment]
It would be nice to add an url to a readme (markdown) to the package:

```yaml
name: Dummy App
type: desktop-application # not sure we need that
categories:
- Media
custom:
- repo: https://github.com//johndoe/harbour-dummy-app/
icon: https://raw.githubusercontent.com//johndoe/harbour-dummy-app/master/harbour-dummy-app.svg
screenshots:
- https://github.com/johndoe/harbour-dummy-app/raw/master/metadata/screenshots/screenshot1.png
- https://github.com//johndoe/harbour-dummy-app/raw/master/metadata/screenshots/screenshot2.png
- https://github.com//johndoe/harbour-dummy-app/raw/master/metadata/screenshots/screenshot3.png
url:
- donation: https://www.paypal.me/johndoe
- readme: https://github.com/johndoe/harbour-dummy-app/raw/master/README.md
```
This could be rendered in Chum GUI instead of the description if present.

**Comment by rinigus** [comment]
@black-sheep-dev: I have opened an issue at https://github.com/sailfishos-chum/sailfishos-chum-gui/issues/73 , will reply there.

**Comment by Olf0** [comment]
All tasks [in the original post](https://github.com/sailfishos-chum/main/issues/29#issue-1028359666) of the first section "app" are fulfilled by the [SailfishOS:Chum GUI application](https://github.com/sailfishos-chum/sailfishos-chum-gui) (hence I ticked these checkboxes), but for the section "web", the SailfishOS-OBS only provides the aspect "search" by its search field for [the SailfishOS:Chum community repository](https://build.merproject.org/project/show/sailfishos:chum).

Still I believe that is sufficient and this issue can be closed.

**Comment by rinigus** [comment]
Agreed, it is sufficient. Closing.

**Comment by HenkKalkwater** [comment]
Sorry for bumping a closed issue, but I have been working on a web interface for Chum, if you are interested. You can find [its source code in this repo](https://github.com/HenkKalkwater/chum-web) and [the website running over here](https://henkkalkwater.github.io/chum-web/).

It is a static website generator written in Python, that runs every other day in GitHub actions and publishes the result to GitHub pages. The generated website lists all apps and packages by name and category, shows recently updated packages and has search functionality. 

There are surely some improvements that could be made regarding functionality, design and others. If you have any suggestions, feel free to open an issue on the repo. Especially for the [about page](https://henkkalkwater.github.io/chum-web/about.html), since I mainly copied it from [this README](https://github.com/sailfishos-chum/main#readme)

I'd be interested in transferring it to the SailfishOS:Chum organisation if that's okay, so that the site can be published under the [sailfishos-chum.github.io](https://sailfishos-chum.github.io) domain and so that the notice at the top of the site can be removed.

**Comment by poetaster** [comment]
> It is a static website generator written in Python, that runs every other day in GitHub actions and publishes the result to GitHub pages. The generated website lists all apps and packages by name and category, shows recently updated packages and has search functionality.

Neato! No hosting required :)

> I'd be interested in transferring it to the SailfishOS:Chum organisation if that's okay, so that the site can be published under the [sailfishos-chum.github.io](https://sailfishos-chum.github.io) domain and so that the notice at the top of the site can be removed.

I think this is a great idea.

**Comment by piggz** [comment]
I too think this is a great idea.  How are you determining recent  updates?  Can you order by update?  Could you also provide an API for the native app to order by update date?

**Comment by HenkKalkwater** [comment]
> I too think this is a great idea. How are you determining recent updates? Can you order by update? Could you also provide an API for the native app to order by update date?

In the `primary.xml.gz` file of the repository, I use the time in the `file` attribute of the `time` element in the `package` element. I assume this timestamp represents when the source to build the package was updated, but since I can't find any formal documentation of the format, it is just a guess. The best documentation I found was https://en.opensuse.org/openSUSE:Standards_Rpm_Metadata.
```xml
<metadata xmlns="http://linux.duke.edu/metadata/common" xmlns:suse="http://novell.com/package/metadata/suse/common" xmlns:rpm="http://linux.duke.edu/metadata/rpm" packages="1331">
    <package type="rpm">
        <!-- most properties omitted -->
        <time file="1694576425" build="1694576388"/>
    </package>
    <!-- Packages omitted -->
</metadata>
```

This seems to work well enough for now, but I suspect whenever a new repository is set up for a new Sailfish OS version, the time for all packages will be set to a little while after the new repository has been created.

**Comment by rinigus** [comment]
@HenkKalkwater, great job! Indeed, it would be great to have it under this ORG. We will just make you an owner of that repository and you could continue your work on it without any problems. Reopening an issue, as it seems to be still closed

**Comment by HenkKalkwater** [comment]
@rinigus I need to transfer the repository to someone who was the permission to create public repositories in the sailfishos-chum organisation. Can I transfer it to you, so you can transfer it to the organisation and add me back as collaborator?

**Comment by rinigus** [comment]
Hmm, I think you can transfer directly. Try and we can see if it works

**Comment by HenkKalkwater** [comment]
I get an error that I don’t have the permission to create public repositories on sailfishos-chum

**Comment by rinigus** [comment]
Then we will make it easier - I'll import it :). One sec

**Comment by rinigus** [comment]
Its currently as chum-web. When ready, we should rename it to something that goes into URL proposed by you. @HenkKalkwater, you should get an invite from Github for that repo

**Comment by HenkKalkwater** [comment]
Okay, I got it deployed on https://sailfishos-chum.github.io/ and I have removed the notice at the top!

**Comment by rinigus** [comment]
Congratulations! Closing here again and please announce it on the forum :)

**Comment by HenkKalkwater** [comment]
Post on the formum: https://forum.sailfishos.org/t/announcing-chum-web/17593?u=ahappyhuman

Op Donderdag 14 december 2023 schreef rinigus:

**Comment by HenkKalkwater** [comment]
@piggz 
> I too think this is a great idea. How are you determining recent updates? Can you order by update? Could you also provide an API for the native app to order by update date?

If the information is accessible from within the library you within chum-gui, I suppose that would have preference. Alternatively, I can provide either a RSS feed with the latest updates [as an user requested](https://github.com/sailfishos-chum/sailfishos-chum.github.io/issues/1) or something like the [package.json](https://sailfishos-chum.github.io/packages.json) file that is currently generated to power the search results, but then sorted by update time.

What would have your preference?

**Comment by Olf0** [comment]
I took the freedom to mark the last three un-ckecked tasks as done, as https://sailfishos-chum.github.io/ provides these features.
